.github/workflows/operator-integration-test.yml

```diff
@@ -0,0 +1,90 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+name: operator Integration Test
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [ opened, reopened ]
+    branches:
+      - main
+  workflow_dispatch:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  AmazonCloudWatchAgentOperatorIntegrationTest:
+    name: AmazonCloudWatchAgentOperatorIntegrationTest
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Start minikube
+        uses: medyagh/setup-minikube@master
+
+      - name: Deploy cert-manager to minikube
+        run:
+          kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.12.0/cert-manager.yaml
+
+      - name: Verify minikube and cert-manager
+        run: |
+          sleep 10
+          kubectl get pods -A
+
+      - name: Build image
+        run: |
+          eval $(minikube docker-env)
+          make container
+          docker images
+
+      - name: Deploy operator to minikube
+        run:
+          make deploy
+
+      - name: Test case for AmagonCloudWatchAgent pod creation
+        run: |
+          helm template --namespace amazon-cloudwatch -s templates/cloudwatch-agent-daemonset.yaml ./helm | kubectl apply --namespace amazon-cloudwatch -f -
+          sleep 60
+          kubectl describe pods -n amazon-cloudwatch
+          pod_name="$(kubectl get pods -n amazon-cloudwatch -l app.kubernetes.io/component=amazon-cloudwatch-agent,app.kubernetes.io/instance=amazon-cloudwatch.cloudwatch-agent -o=jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}')"
+          if [ -z "$pod_name" ]; then
+            echo "Pod $pod_name is not running. Exiting with ERROR."
+            echo "Pod is not created test case: FAIL"
+          else
+            echo "Pod $pod_name is running."
+            echo "Pod is created test case: PASS"
+          fi
+
+      - name: Test default instrumentation
+        run: |
+          cd integration-tests/operator
+          kubectl apply -f ./test-deployment.yaml
+          sleep 60
+          kubectl get pods -A 
+          kubectl describe pods -n default
+          cd ./cmd
+          go run validate_instrumentation_vars.go default default_instrumentation_env_variables.json
```

Comment:
We are introducing a new pattern here that is diff from all the other integ tests we have currently that leverage tf. Would have been nice to just stay consistent.

Reply:
I think this test is more lightweight so that we do not have to use terraform.

This is a test that can be used on every PR that spans about 6 minutes or so to quickly test the functionality of the operator.

We have the EKS / helm / E2E test that fully tests the whole system which uses terraform.

I like having the simplicity here imo.

Reply:
Doing it via terraform shouldnt change the runtime (or atleast by a significant amount). We can use the same minikube resource with terraform if we want to avoid bringing up an EKS cluster. But doing so already has comprises like being unable to verify if the agent is functional.

I think we are very shortly going to add health checks so itll be interesting how we handle that then.

Reply:
that's fair, we can use a linux ec2 for example.

this actually brings up a good point though -- is cwa only supported on AWS resources? I thought it would be able to run on native k8s?

Reply:
It should be able to run on native k8s as well with the right creds config setup